### ISSUE TITLE
LSTM weights not loaded correctly in tf.keras

### ISSUE THREAD

**Issue by ptiwald** [tensorflow/tensorflow#23013]
### Description
Running into this problem when I:
- build model A containing an LSTM layer
- save the model weights with `tf.keras.save_weights()` in TensorFlow format
- at some later point create an identical model A' and load the previously saved weights

Inspection by eye shows the lstm weights differ between model A and A' while weights of other layers (e.g. dense) are loaded correctly from file. When running through the same procedure as above but saving the weights in HDF5 format everything works as expected.

### System Information:
- Have I written custom code: yes
- OS Platform and Distribution: Linux Ubuntu 16.04
- Mobile device: NA
- TensorFlow installed from (source or binary): binary (pip package)
- TensorFlow version: 1.10.0
- Python version: 3.6.6
- Bazel version: NA
- GCC/Compiler version: NA
- CUDA/cuDNN version: NA
- GPU model and memory: NA
- Exact command to reproduce:
```
import tensorflow as tf

input_layer = tf.keras.layers.Input(shape=(None, 5))
lstm_layer = tf.keras.layers.LSTM(units=2)(input_layer)
output_layer = tf.keras.layers.Dense(units=2, activation='softmax')(lstm_layer)

my_model = tf.keras.Model(input_layer, output_layer)

my_model.save_weights("weights_test/my_weights")

print('\nlstm:')
print(my_model.layers[1].get_weights())
print('\ndense:')
print(my_model.layers[2].get_weights())

#lstm:
#[array([[ 0.38913965,  0.4081726 , -0.23971727,  ...
#dense:
#[array([[-0.5567662 , -0.38940358],
#       [ 0.16435587,  1.1092712 ]], dtype=float32), array([0., 0.], dtype=float32)]

new_input_layer = tf.keras.layers.Input(shape=(None, 5))
new_lstm_layer = tf.keras.layers.LSTM(units=2)(new_input_layer)
new_output_layer = tf.keras.layers.Dense(units=2, activation='softmax')(new_lstm_layer)

my_new_model = tf.keras.Model(new_input_layer, new_output_layer)
my_new_model.load_weights("weights_test/my_weights")

print('\nlstm:')
print(my_new_model.layers[1].get_weights())
print('\ndense:')
print(my_new_model.layers[2].get_weights())

#lstm:
#[array([[-0.41712055,  0.67933106, -0.5161066 , ...
#dense:
#[array([[-0.5567662 , -0.38940358],
#       [ 0.16435587,  1.1092712 ]], dtype=float32), array([0., 0.], dtype=float32)]
```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by ptiwald** [comment]
just added the System Information to the issue

**Comment by ptiwald** [comment]
@yongtang  Tried different versions of tensorflow. Seems the bug is fixed from 1.11. onward. Thanks and sorry for not checking.

**Comment by yongtang** [comment]
@ptiwald Glad it works. 👍 I will closing this issue, but feel free to reopen if there is still issues related.
